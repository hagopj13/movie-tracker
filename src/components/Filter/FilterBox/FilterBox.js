// @flow
import React from 'react';
import type { Node } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';

type Props = {
  children: Node,
  title: string,
};

const useStyles = makeStyles((theme) => ({
  root: {
    minWidth: 270,
  },
  title: {
    fontWeight: theme.typography.fontWeightBold,
    padding: theme.spacing(1, 2),
  },
  filters: {
    diplay: 'flex',
    flexDirection: 'column',
  },
}));

const FilterBox = (props: Props) => {
  const { children, title } = props;

  const classes = useStyles();

  return (
    <Paper className={classes.root} elevation={3}>
      <Typography className={classes.title} variant="h6">
        {title}
      </Typography>
      <Divider />
      <div className={classes.filters}>{children}</div>
    </Paper>
  );
};

export default FilterBox;
