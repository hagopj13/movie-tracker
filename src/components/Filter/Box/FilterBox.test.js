import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import Typography from '@material-ui/core/Typography';

import FilterBox from './FilterBox';

describe('FilterBox component', () => {
  let shallow;
  let title;
  let wrapper;

  const ChildComponent = () => <div>Child component</div>;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    title = 'Some title';
    const mockProps = {
      title,
    };
    wrapper = shallow(
      <FilterBox {...mockProps}>
        <ChildComponent />
      </FilterBox>,
    );
  });

  it('should render the FilterBox component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the title', () => {
    expect(wrapper.find(Typography).text()).toBe(title);
  });

  it('should render the child component', () => {
    expect(wrapper.exists(ChildComponent)).toBe(true);
  });
});
