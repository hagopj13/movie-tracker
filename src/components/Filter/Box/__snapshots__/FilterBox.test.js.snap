// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBox component should render the FilterBox component correctly 1`] = `
<WithStyles(ForwardRef(Paper))
  className="makeStyles-root-1"
  elevation={3}
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-title-2"
    variant="h6"
  >
    Some title
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Divider)) />
  <div
    className="makeStyles-filters-3"
  >
    <ChildComponent />
  </div>
</WithStyles(ForwardRef(Paper))>
`;
